packages/client/src/client.d.ts

```diff
@@ -46,5 +46,8 @@ declare class Client {
   request(data: ClientRequest, cb?: (err: ResponseError, response: [ClientResponse, any]) => void): Promise<[ClientResponse, any]>;
 }
 
-declare const client: Client & { Client: typeof Client };
+declare const client: Client;
+// @ts-ignore
```

Comment:
This matches style of mail.d.ts file. Part that I wanted to change here mainly was the `{ Client: typeof Client }` piece. I get the feeling that the goal there was to expose the class as a type, but we do `typeof` on the class itself, instead of an instance of it, so it ends up being a type with just a `prototype` field. Let me know if I'm missing something here.

Reply:
I believe this change might break folks: https://github.com/sendgrid/sendgrid-nodejs/issues/986

Reply:
yeah the goal here was to export `Client` as not just a value with type `typeof Client`, but to export the class itself, which could be used as a value or a type. im pretty sure this will only enhance the existing functionality. currently, users could do:
`const a = new Client()`

but the only way they could pass the actual type around is by doing a default import of the singleton client and then doing a `typeof` on it.

am i missing something here?

Reply:
I don't quite understand why this method is better than what we have. Perhaps a good way to illustrate is to add a sample quick start [here](https://github.com/sendgrid/sendgrid-nodejs/tree/main/docs/use-cases).

Reply:
To give a quick code example:

```
import { Client } from '@sendgrid/client';

// This is possible in both versions
const client = new Client();

// This (I think) is only possible in the new version since Client is only a value in the old version and not a type
function somethingThatTakesInAClient(client: Client) {
...
}
```

Reply:
while it isn't necessary, you do it for most of your other types that I've seen, and if you are going to export functionality, it seems to me it would also make sense to export the type

Reply:
to further illustrate, in the current version, we could accomplish the same by doing:
```
// Note we are importing the singleton client exported by the lib as well as the class itself (as a value)
import client, { Client } from '@sendgrid/client';

// We can do a typeof on the singleton instance. This is equivalent to what is in the last example. For this one you need // to import a singleton instance of the client just to get the type
function somethingThatTakesInAClient(arg: typeof client) {
...
}

// This compiles, but the behavior isn't what you expect. Since we are doing typeof on the class, the only members are // things on the function prototype, so we can't access methods like 'setApiKey' for example
function anotherThingThatLooksLikeItTakesInAClientButDoesntReally(arg: typeof Client) {
...
}
```

Reply:
Here's an example where you use the proposed pattern: https://github.com/sendgrid/sendgrid-nodejs/blob/main/packages/mail/src/mail.d.ts#L39-L42.